src/vos/vos_obj_index.c

```diff
@@ -252,9 +252,10 @@ vos_oi_find(struct vos_container *cont, daos_unit_oid_t oid,
 		ilog = &obj->vo_ilog;
 	}
 
-	tmprc = vos_ilog_ts_add(ts_set, ilog, &oid, sizeof(oid));
-
-	D_ASSERT(tmprc == 0); /* Non-zero return for akey only */
+	if (ts_set) {
```

Comment:
It seems vos_ilog_ts_add() will handle that internally.

Reply:
ok, I'll remove this check.